hooks/useProtectedRoute.ts

```diff
@@ -14,23 +14,25 @@ export function useProtectedRoute({
   redirect = '',
   timeout = -1,
 }: ProtectedProps) {
-  const { checkLoggedIn, status, metadata, magicMetadata } = useLogin({
+  const $router = useRouter()
+  const loginState = useLogin({
```

Comment:
This is part of the issue -- useProtectedRoute is establishing its own login state store by calling `useLogin` instead of using the LoginContext. If clients are reading from it, they'll have a different view of the login state.